upgrade/scripts/k8s/upgrade_k8s.sh

```diff
@@ -0,0 +1,328 @@
+      # Mark the node ready for scheduling.
+      echo "$(prefix) Uncordoning node: [${host}]."
+      kubectl uncordon "${host}"
+    fi
+  else
+    echo "$(prefix) INFO Version ${minor_target_version} is less than ${current_version}. Don't need to upgrade kubelet for '$host'."
+  fi
+done
+
+echo "$(prefix) Upgrading kubelet on worker nodes to ${target_version}."
+
+for host in ${workers}; do
+  # Check current kubelet version. If less than 
+  current_version=$(ssh $host kubelet --version | awk '{ print $2 }' | cut -c 2-5)
+  if [[ $(bc -l <<< "${minor_target_version} > ${current_version}") -eq 1 ]]; then
+    echo "$(prefix) Installing [kubeadm-${target_version}] on [${host}]."
+    ssh "${host}" zypper --non-interactive install "kubeadm-${target_version}"
+    ssh "${host}" zypper --non-interactive install "kubernetes-cni-${k8s_cni[${minor_target_version}]}"
+    ssh "${host}" kubeadm upgrade node
+    kubectl drain --ignore-daemonsets --delete-emptydir-data "${host}"
+
+    # Fix /etc/sysconfig/kubelet
+    fix_sysconfig "${host}"
+    # Remove --container-runtime=remote argument from KUBELET_KUBEADM_ARGS.
+    kube_kubeadm_args='KUBELET_KUBEADM_ARGS="--container-runtime-endpoint=/run/containerd/containerd.sock --pod-infra-container-image=artifactory.algol60.net/csm-docker/stable/k8s.gcr.io/pause:'${pause[${minor_target_version}]}'"'
+    echo "$kube_kubeadm_args" > /var/lib/kubelet/kubeadm-flags.env
+
+    # Update pause version in containerd
+    pause_version="pause:${pause[${minor_target_version}]}"
+    ssh "${host}" sed -i -e "s/pause:3.9/$pause_version/g" /etc/containerd/config.toml
+    # Restart containerd.
+    ssh "${host}" systemctl restart containerd
+
+    # Update kubelet and kubectl. We don't need to update these one at a
+    # time; we can skip up to three versions at a time.
+    echo "$(prefix) Installing [kubelet-${target_version}] and [kubectl-${target_version}] on [${host}]."
+    ssh "${host}" zypper --non-interactive install "kubelet-${target_version}" "kubectl-${target_version}"
+
+    # Reload daemons and restart kubelet.
+    ssh "${host}" systemctl daemon-reload
+    ssh "${host}" systemctl restart kubelet
+
+    kubectl uncordon "${host}"
+
+    # Need to wait 2 minutes for pods to settle before doing another drain.
+    echo "$(prefix) Waiting 2 minutes for pods to settle ..."
+    sleep 120
+  else
+    echo "$(prefix) INFO Version ${minor_target_version} is less than ${current_version}. Don't need to upgrade kubelet for '$host'."
+  fi
+done
+
+echo "$(prefix) Upgrade to ${target_version} complete."
+
+echo "$(prefix) Verifying that all ncn nodes at expected K8s ${v_target_version}"
+all_ncns="${masters} ${workers}"
+BAD_NODES=0
+for host in ${all_ncns}; do
+  # Check kubeadm version on all nodes and make sure it matches v_target_version
+  current_version=$(ssh $host kubeadm version -o json | jq -r '.clientVersion.gitVersion')
+  current_minor_version=$(echo "$current_version" | cut -c 2-5)
```

Comment:
You can use the shell to strip off the leading `v` and the patch number, though it takes two steps:
```suggestion
  current_minor_version=${current_version%.*}
  current_minor_version=${current_minor_version#v}
```